OpenRA.Mods.Common/Traits/Attack/AttackDetonates.cs

```diff
@@ -77,11 +76,23 @@ public void ResolveOrder(Actor self, Order order)
 					self.CancelActivity();
 
 				self.SetTargetLine(target, Color.Red);
-				self.QueueActivity(new MoveAdjacentTo(self, target));
-				self.QueueActivity(new CallFunc(() => Explode(self)));
+
+				if (move is Mobile)
+					self.QueueActivity(new MoveAdjacentTo(self, target));
+				else
+				{
+					var aircraft = move as Aircraft;
+					if (aircraft != null)
+					{
+						if (aircraft.IsPlane)
+							self.QueueActivity(new Fly(self, target));
+						else
+							self.QueueActivity(new HeliFly(self, target));
+					}
```

Comment:
Could `aircraft.MoveToTarget` be used here? That also seems to account for turns and such.